### DISCUSSION TITLE
how to set underline width? thanks

### DISCUSSION THREAD

**Discussion by dfsung** [sumatrapdfreader/sumatrapdf#4108] [Documentation]
how to set underline width in sumatrapdfreader ? thanks

**Comment by GitHubRulesOK** [comment]
Interesting question and unsure how it may be set other than keep default of 1 here from another EDITOR app is a line at scale 4 so 4 times as thick obliterates the decenders y g p  etc.
![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/c2d96501-024f-4b57-842c-13f999649b85)


If in any other edit (including SumatraPDF) we alter a feature such as colour the thickness drops to the default scale of 1
![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/4d87fe2f-0661-440c-93ef-bb12939c0ea4)

![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/b4dfcf27-cc79-4651-a1ec-855e7f72f2c7)

**Reply by dfsung** [reply]
So, in sumatrapdfreader, don't set underline width?

**Reply by GitHubRulesOK** [reply]
It is not normal PDF for all documents so the default unit in a PDF is 1/72" unless the file is "scaled" UserUnits so if a file is as large as the planet the Userunits would be thickened by the same amount.  The older standard blurb is below where I have highlighted the constraints

TextDecoration**Thickness**
number
(**Optional**; inheritable; **PDF 1.5**) The thickness of each line drawn as
part of the text decoration. The **value shall be a non-negative
number in default user space units representing the thickness** (0 is
interpreted as the thinnest possible line). If this attribute is not
specified, **it shall be derived from the current stroke thickness in
effect at the start of the element’s associated content, transformed
into default user space units.**
TextDecorationType
name
(Optional; not inheritable) The text decoration, if any, to be applied
to the element’s text.
None
No text decoration
**Underline
A line below the text**
Overline
A line above the text
LineThrough
A line through the middle of the text
Default value: None.
**This attribute shall apply to all text content items that are children of
this element** or of its nested ILSEs, if any. The attribute shall not
apply to nested BLSEs or to content items other than text.
**The colour, position, and thickness of the decoration shall be
uniform across all children, regardless of changes in colour, font
size, or other variations in the content’s text characteristics.**


So as with many PDF features it is non prescriptive as say 10% or 1 unit thick
but the common default is usually 1 point thick to not obliterate the character outline of 4 points and above
